bin/comment_on_pr

```diff
@@ -1,14 +1,125 @@
 #!/bin/bash -eu
 
-PR_MESSAGE_FILE=$1
-GITHUB_TOKEN=$2
+# This script is used to comment on a pull request, and must be run in a buldkite PR step.
+#
+# Usage:
+#
+#    # Post a new comment to the PR
+#    comment_on_pr "This is a comment" <github-token>
+#
+#    # Post a new comment which can be uniqly identifiered by the given comment id,
+#    # or update the existing comment that's associated with the given comment id.
+#    # You can choose to delete the existing comment if it exists by passing `--if-exist delete`.
+#    comment_on_pr --id comment-id "This is a comment" [--if-exist update|delete] <github-token>
+#
+#    # Delete the comment associated with the given comment id
+#    comment_on_pr --id comment-id <github-token>
+#
+# Please note the comment argument can be a path to a file containing the comment, or a markdown.
 
-if ! command -v gh &> /dev/null; then
-	brew install gh
+# Check dependencies and print their versions for diagnosis purposes
+cat <<EOF
+> jq --version
+$(jq --version)
+
+> curl --version
+$(curl --version)
+
+EOF
+
+if [[ "${BUILDKITE_PULL_REQUEST:-invalid}" == "invalid" ]]; then
+    echo "Error: this tool can only be called from a Buildkite PR job"
+    exit 1
+fi
+
+# Default options
+opt_comment_id=""
+opt_if_exist="update"
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --id) opt_comment_id="$2"; shift ;;
+        --if-exist) opt_if_exist="$2"; shift ;;
+        --*) echo "Unknown option: $1"; exit 1 ;;
+        *) break ;;
+    esac
+    shift
+done
+
+case "$#" in
+    1)
+        arg_pr_comment=""
+        arg_github_token=$1
+        ;;
+    2)
+        if [[ -f "$1" ]]; then
+            arg_pr_comment=$(cat "$1")
+        else
+            arg_pr_comment="$1"
+        fi
+
+        arg_github_token=$2
+        ;;
+    *) echo "Error: invalid comment argument"; exit 1 ;;
+esac
+
+github_user=$(basename "$(dirname "${BUILDKITE_PULL_REQUEST_REPO%.git}")")
+github_repo=$(basename "${BUILDKITE_PULL_REQUEST_REPO%.git}")
+issues_endpoint="https://api.github.com/repos/${github_user}/${github_repo}/issues"
+
+# Find the existing PR comment
+if [[ -n "$opt_comment_id" ]]; then
+    # Do not change this variable value, it's used to identify the comments posted by this utility.
+    comment_body_id="<!-- DO NOT REMOVE ci-toolkit-comment-identifier: $opt_comment_id -->"
+    existing_comment_id="$(curl -s \
+        -H "Authorization: token ${arg_github_token}" \
+        -H "Accept: application/vnd.github+json" \
+        "$issues_endpoint/${BUILDKITE_PULL_REQUEST}/comments" \
+        | jq ".[] | select(.body | contains(\"$comment_body_id\")) | .id")"
```

Comment:
I've added a step to valid the token by making an API request: https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/51/commits/88ff81de10b3867d8d522109162a897cddd18e36

Reply:
> it would also be nice to have arg_github_token pick up the existing GITHUB_TOKEN env var by default if not provided explicitly

Done in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/51/commits/11671dd87536d4e65444eb5b843a0c70d415d1f0